src/plugins/kibana_utils/public/state_containers/create_state_container_react_helpers.ts

```diff
@@ -35,7 +35,7 @@ export const createStateContainerReactHelpers = <Container extends StateContaine
     return value;
   };
 
-  const useTransitions = () => useContainer().transitions;
+  const useTransitions: () => Container['transitions'] = () => useContainer().transitions;
```

Comment:
This allowed to remove this type casting: https://github.com/elastic/kibana/pull/54436/files#diff-0474422dfd150c2b85e0efb48c007a20L196